File: rules/gitlab/scala/crypto/test-BlowfishKeySize.scala
// License: LGPL-3.0 License (c) find-sec-bugs
package crypto

import javax.crypto.KeyGenerator
import java.security.NoSuchAlgorithmException

class BlowfishKeySize {
  @throws[NoSuchAlgorithmException]
  def danger(): Unit = {
    val keyGen = KeyGenerator.getInstance("Blowfish")
    keyGen.init(64)
  }
}

File: rules/gitlab/java/crypto/test-BlowfishKeySize.java
// License: LGPL-3.0 License (c) find-sec-bugs
package crypto;

import javax.crypto.KeyGenerator;
import java.security.NoSuchAlgorithmException;

public class BlowfishKeySize {
    public void danger() throws NoSuchAlgorithmException {
        KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
        keyGen.init(64);
    }
}
